internal/report/source.go

```diff
@@ -347,20 +347,40 @@ func printFunctionHeader(w io.Writer, name, path string, flatSum, cumSum int64,
 		measurement.Percentage(cumSum, rpt.total))
 }
 
+func calculatePercentile(partial, sum int64) string {
+	percentile := partial * 100 / sum
```

Comment:
This is not percentile.

Reply:
Will change this. Misunderstood its meaning.

Reply:
Quick question: if we use the percentile, when considering the different lines in a function, it could happen that a lot of them will have a 0 value. Given the fact that this results in a bunch of duplicates, should I run the percentile on a set of these values? 

Including dulicates could make lines with 0 contribution be part of the 20th percentile or worse.

Reply:
I have now implemented percentile. I'm not sure if my code (in terms of style and correctness) is fine, thus I have pushed a first version without tests, to get it reviewed. I will add tests soon. 

I wonder if the percentiles functions should go into a different file or if they belong where they are.